hyperspace/cosmos/src/utils.rs

```diff
@@ -0,0 +1,485 @@
+use crate::error::Error;
```

Comment:
I think most of the necessary functions to extract/filter/converts events exist in the following sections, in particular in the [ics07-tendermint/src/events.rs](https://github.com/ComposableFi/centauri/blob/master/light-clients/ics07-tendermint/src/events.rs). But we have to refactor it and import some others from `ibc-module` so that it can be called in different places for the methods of `IbcProvider` trait  

- [ibc module events](https://github.com/ComposableFi/centauri/blob/master/ibc/modules/src/events.rs
)
- [ics02_client/events
](https://github.com/ComposableFi/centauri/blob/master/ibc/modules/src/core/ics02_client/events.rs
)
- [ics03_connection/events](https://github.com/ComposableFi/centauri/blob/master/ibc/modules/src/core/ics03_connection/events.rs
)
- [ics04_channel/events](https://github.com/ComposableFi/centauri/blob/master/ibc/modules/src/core/ics04_channel/events.rs)

Reply:
yeah, we'll resolve that in another PR.